### DISCUSSION TITLE
Publicando pacotes com dependências privadas

### DISCUSSION THREAD

**Discussion by saulomlcosta** [balta-io/forum#332] [Tira Dúvidas] [answered]
Curso: Fundamentos do Azure, Git, GitHub e DevOps 
Aula:  Publicando pacotes com dependências privadas

Ao tentar fazer o build do "SecondPrivatePackage", que contem referencia do primeiro pacote já importado, está me acontecendo um erro relacionado a autorização. Basicamente, ele tenta acessar o primeiroPacote, porém tem acesso negado, no momento estou vendo as configs do Github.

Meu YML
`name: Build and deploy package

on:
  push:
    branches:
      - main

jobs:
  build-and-deploy:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@main

      - name: Set up .NET Core
        uses: actions/setup-dotnet@v2
        with:
          dotnet-version: "7.0.x"

      - name: Add GitHub Packages Source
        run: dotnet nuget add source https://nuget.pkg.github.com/saulomlcosta/index.json -n github -u saulomlcosta -p ${{ secrets.GITHUB_TOKEN }} --store-password-in-clear-text
    
      - name: Build with dotnet
        run: dotnet build --configuration Release

      - name: Create the package
        run: dotnet pack --configuration Release

      - name: Publish
        run: dotnet nuget push "bin/Release/*.nupkg" -k ${{ secrets.GITHUB_TOKEN }} -s https://nuget.pkg.github.com/saulomlcosta/index.json --skip-duplicate
`

Erro no Workflow
![image](https://github.com/balta-io/forum/assets/79532045/d9cd6fb9-c938-49ff-ba01-7f6d8c80e892)

**Comment by andrebaltieri** [comment, marked as answer]
@saulomlcosta bom dia, por favor, dá uma conferida nestes dois tópicos:
- https://github.com/balta-io/forum/discussions/315
- https://github.com/balta-io/forum/discussions/84

**Reply by saulomlcosta** [reply]
Vou ver sim, brigadão.

**Comment by saulomlcosta** [comment]
@andrebaltieri mestre Balta, deu tudo certo. Basicamente, ele não estava conseguindo acessar o ${{ secrets.GITHUB_TOKEN}}. Para resolver, segui sua ideia utilizada com o NUGET, para guardar o NUGET_TOKEN. Porém com o token do Github. E tudo certo, tanto build quanto publicação. Obrigado pelo retorno e pela indicação de posts.

![image](https://github.com/balta-io/forum/assets/79532045/f3e26223-f5dd-4b54-a18e-7d4114a68893)

![image](https://github.com/balta-io/forum/assets/79532045/988ded3c-107d-422c-be25-900f3c5989cc)
